src/Utilities/Protocols/StaticReturnApplyable.hpp

```diff
@@ -0,0 +1,46 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <type_traits>
+
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TypeTraits/IsA.hpp"
+
+namespace protocols {
+/// Protocol for a struct with a static `apply` function returning and
+/// taking arguments based on tags in `return_tags` and
+/// `argument_tags` type aliases.
+///
+/// \snippet Test_StaticReturnApplyable.cpp StaticReturnApplyable
+struct StaticReturnApplyable {
+  template <typename ConformingType>
+  struct test {
+    using argument_tags = typename ConformingType::argument_tags;
+    using return_tags = typename ConformingType::return_tags;
+    static_assert(tt::is_a_v<tmpl::list, argument_tags>);
+    static_assert(tt::is_a_v<tmpl::list, return_tags>);
+
+    static_assert(
+        std::is_same_v<return_tags, tmpl::remove_duplicates<return_tags>>,
+        "return_tags should not contain duplicates.");
+    // Duplicates in argument_tags are OK.
+
+    static_assert(
+        tmpl::wrap<
+            tmpl::append<
+                tmpl::list<decltype(ConformingType::apply)>,
```

Comment:
After having a conversation with @nilsvu, he mentioned this shouldn't when BaseTags are present, since they do not know their type.  I am going to proceed by keeping the first three `static_asserts` that check for argument_tags, return_tags, and no duplicate return_tags. That should work independently of templated or nontemplated apply functions.  What are your thoughts?

Reply:
Good point.  That's sounds good.